packages/scm/src/browser/scm-tree-widget.tsx

```diff
@@ -324,11 +319,7 @@ export class ScmTreeWidget extends TreeWidget {
         if (!repository) {
             return;
         }
-        const parentNode = node.parent;
-        if (!(parentNode && (ScmFileChangeFolderNode.is(parentNode) || ScmFileChangeGroupNode.is(parentNode)))) {
-            return;
-        }
-        const groupId = parentNode.groupId;
+        const groupId = this.getGroupIdForFileNode(node);
         const group = repository.provider.groups.find(g => g.id === groupId)!;
         return group.resources.find(r => String(r.sourceUri) === node.sourceUri)!;
```

Comment:
getGroupIdForFileNode should never actually return '' as that would mean we have an invalid model.  I have changed it to throw an Error instead (and also moved to the node's namespace so it can be reused in one more place).